src/app/shared/Compose.svelte

```diff
@@ -124,27 +151,38 @@
     }
 
     // Enter adds a newline, so do it on key down
-    if (["Enter"].includes(e.code)) {
-      autocomplete({pubkey: suggestions.get()})
-    }
-
-    // Only autocomplete topics on space
-    if (["Space"].includes(e.code)) {
-      if (autocomplete()) {
-        e.preventDefault()
+    if (["Enter"].includes(e.code) && suggestions.get()) {
+      e.preventDefault()
+      const pubkey = suggestions.get()
+      if (pubkey) {
+        const content = $editor.getHTML()
+        // only replace the last occurence
+        const replacedContent = content.replace(/(.*)@\w+/, "$1")
+        $editor.commands.setContent(replacedContent, true)
+        $editor.commands.insertNProfile({nprofile: pubkeyEncoder.encode(pubkey)})
+        suggestions.setData([])
```

Comment:
This approach will probably not work, tiptap is too magical, you have to do things their way.